app/uk/gov/hmrc/agentservicesaccount/controllers/SuspendedJourneyController.scala

```diff
@@ -70,10 +72,62 @@ class SuspendedJourneyController @Inject() (
           _ <- cacheService.put(NAME, formData.name)
           _ <- cacheService.put(EMAIL, formData.email)
           _ <- cacheService.put(PHONE, formData.phone)
-          _ <- cacheService.put(UTR, formData.utr.getOrElse(""))
           _ <- cacheService.put(ARN, request.agentInfo.arn.value)
-        } yield Redirect("")
+        } yield Redirect(routes.SuspendedJourneyController.showSuspendedDescription())
       }
     )
   }
+
+  def showSuspendedDescription: Action[AnyContent] = actions.authActionOnlyForSuspended.async { implicit request =>
+    cacheService.get(DESCRIPTION).flatMap {
+      case Some(description) => Ok(recoveryDescriptionView(SuspendDescriptionForm.form.fill(description))).toFuture
+      case _ => Ok(recoveryDescriptionView(SuspendDescriptionForm.form)).toFuture
+    }
+  }
+  def submitSuspendedDescription: Action[AnyContent] = actions.authActionOnlyForSuspended.async { implicit request =>
+    SuspendDescriptionForm.form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => recoveryDescriptionView(formWithErrors).toFuture
+          .map(BadRequest(_)),
+        formData => {
+          for {
+            _ <- cacheService.put(DESCRIPTION, formData)
+          } yield Redirect(routes.SuspendedJourneyController.showSuspendedSummary())
+        }
+      )
+  }
+
+  private def getSummaryDetails (implicit request: Request[_]) = {
+  for {
+     name <- cacheService.get(NAME)
+     email <- cacheService.get(EMAIL)
+     phone <- cacheService.get(PHONE)
+     description <- cacheService.get(DESCRIPTION)
+     arn <- cacheService.get(ARN)
+  } yield (name,email,phone, description, arn) match {
+    case (Some(n), Some(e),Some(p), Some(d), Some(a)) =>
+      Option(AccountRecoverySummary(n,e,p,d,a))
+    case _ => None
+  }
+}
+  def showSuspendedSummary: Action[AnyContent] = actions.authActionOnlyForSuspended.async { implicit request =>
+    getSummaryDetails.flatMap{
+      case Some(summaryDetails) => Ok(checkRecoveryAnswersView(summaryDetails)).toFuture
+      case None => Redirect(routes.SuspendedJourneyController.showContactDetails()).toFuture
+    }
+  }
+   def submitSuspendedSummary: Action[AnyContent] = actions.authActionOnlyForSuspended.async { implicit request =>
+
+     getSummaryDetails.flatMap {
+
+     case Some(summaryDetails) => emailService.sendSuspendedSummaryEmail(request.agentInfo.arn,summaryDetails)
```

Comment:
space here too? and what happens if there is None? this should redirect to the start of the journey like the other pages do

Reply:
OK